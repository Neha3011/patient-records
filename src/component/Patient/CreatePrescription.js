import React from 'react';
import _ from 'lodash';

import { normalizeAddPrescriptionData } from '../../utils/normalizer';
import { setToLocalStorage } from '../../utils/localStorageHelpers';

class CreatePrescription extends React.Component {
  state = {
    'medicines': [{}, {}, {}],
    'isSavedPrescription': false
  };

  onMedicinesChange = (key, index, value) => {
    const medicines = _.clone(this.state.medicines);
    medicines[index][key] = value.target.value;
    this.setState({
      medicines
    });
  };

  addPrescription = () => {
    const prescriptions = normalizeAddPrescriptionData(this.state.medicines);
    setToLocalStorage('prescriptions', JSON.stringify(prescriptions));
    this.setState({
      'isSavedPrescription': true
    }, () => {
      setTimeout(() => {
        this.setState({
          'medicines': [{}, {}, {}],
          'isSavedPrescription': false
        });
      }, 2000);
    });
  };

  render() {
    return (
      <div className="patient__actions">
        <div className="patient__actions__create">
          <div className="patient__actions__header">
            Add prescription
          </div>

          <div className="patient__actions__container">
            <div className="patient__actions__create__details">
              Name :
              <input
                type="text"
                placeholder="Enter medicine name here"
                className="patient__actions__create__details__name"
                value={this.state.medicines[0].name || ''}
                onChange={this.onMedicinesChange.bind(this, 'name', 0)}
              />
            </div>

            <div className="patient__actions__create__details">
              Quantity :
              <input
                type="number"
                placeholder="Qty"
                className="patient__actions__create__details__qty"
                value={this.state.medicines[0].qty || ''}
                onChange={this.onMedicinesChange.bind(this, 'qty', 0)}
              />
            </div>
          </div>

          <div className="patient__actions__container">
            <div className="patient__actions__create__details">
              Name :
              <input
                type="text"
                placeholder="Enter medicine name here"
                className="patient__actions__create__details__name"
                value={this.state.medicines[1].name || ''}
                onChange={this.onMedicinesChange.bind(this, 'name', 1)}
              />
            </div>

            <div className="patient__actions__create__details">
              Quantity :
              <input
                type="number"
                placeholder="Qty"
                className="patient__actions__create__details__qty"
                value={this.state.medicines[1].qty || ''}
                onChange={this.onMedicinesChange.bind(this, 'qty', 1)}
              />
            </div>
          </div>

          <div className="patient__actions__container">
            <div className="patient__actions__create__details">
              Name :
              <input
                type="text"
                placeholder="Enter medicine name here"
                className="patient__actions__create__details__name"
                value={this.state.medicines[2].name || ''}
                onChange={this.onMedicinesChange.bind(this, 'name', 2)}
              />
            </div>

            <div className="patient__actions__create__details">
              Quantity :
              <input
                type="number"
                placeholder="Qty"
                className="patient__actions__create__details__qty"
                value={this.state.medicines[2].qty || ''}
                onChange={this.onMedicinesChange.bind(this, 'qty', 2)}
              />
            </div>
          </div>

          {(() => {
            if (this.state.isSavedPrescription) {
              return (
                <div className="patient__success">
                  Saved Successfully
                </div>
              );
            }
          })()}
          <button
            className="patient__actions__submit"
            onClick={this.addPrescription}
          >
            Submit Prescription
          </button>
        </div>
      </div>
    );
  }
}

export default CreatePrescription;
