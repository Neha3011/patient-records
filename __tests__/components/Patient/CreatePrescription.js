import React from 'react';
import CreatePrescription from '../../../src/component/Patient/CreatePrescription';
import renderer from 'react-test-renderer';

test('render Header:', () => {
  const component = renderer.create(
      <CreatePrescription />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});