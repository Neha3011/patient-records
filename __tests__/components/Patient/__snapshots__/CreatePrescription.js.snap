// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Header: 1`] = `
<div
  className="patient__actions"
>
  <div
    className="patient__actions__create"
  >
    <div
      className="patient__actions__header"
    >
      Add prescription
    </div>
    <div
      className="patient__actions__container"
    >
      <div
        className="patient__actions__create__details"
      >
        Name :
        <input
          className="patient__actions__create__details__name"
          onChange={[Function]}
          placeholder="Enter medicine name here"
          type="text"
          value=""
        />
      </div>
      <div
        className="patient__actions__create__details"
      >
        Quantity :
        <input
          className="patient__actions__create__details__qty"
          onChange={[Function]}
          placeholder="Qty"
          type="number"
          value=""
        />
      </div>
    </div>
    <div
      className="patient__actions__container"
    >
      <div
        className="patient__actions__create__details"
      >
        Name :
        <input
          className="patient__actions__create__details__name"
          onChange={[Function]}
          placeholder="Enter medicine name here"
          type="text"
          value=""
        />
      </div>
      <div
        className="patient__actions__create__details"
      >
        Quantity :
        <input
          className="patient__actions__create__details__qty"
          onChange={[Function]}
          placeholder="Qty"
          type="number"
          value=""
        />
      </div>
    </div>
    <div
      className="patient__actions__container"
    >
      <div
        className="patient__actions__create__details"
      >
        Name :
        <input
          className="patient__actions__create__details__name"
          onChange={[Function]}
          placeholder="Enter medicine name here"
          type="text"
          value=""
        />
      </div>
      <div
        className="patient__actions__create__details"
      >
        Quantity :
        <input
          className="patient__actions__create__details__qty"
          onChange={[Function]}
          placeholder="Qty"
          type="number"
          value=""
        />
      </div>
    </div>
    <button
      className="patient__actions__submit"
      onClick={[Function]}
    >
      Submit Prescription
    </button>
  </div>
</div>
`;
